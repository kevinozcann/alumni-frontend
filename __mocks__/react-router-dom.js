import React from 'react';

const rrd = jest.genMockFromModule('react-router-dom');

rrd.BrowserRouter = ({ children }) => <div>{children}</div>;

module.exports = rrd;
