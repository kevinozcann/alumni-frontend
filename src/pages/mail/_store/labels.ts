import { IMailLabel } from 'pages/mail/mail-types';

const labels: IMailLabel[] = [
  {
    id: 'inbox',
    type: 'system_label',
    name: 'inbox',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'starred',
    type: 'system_label',
    name: 'starred',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'sent',
    type: 'system_label',
    name: 'sent',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'drafts',
    type: 'system_label',
    name: 'drafts',
    unreadCount: 0,
    totalCount: 0
  },
  {
    id: 'trash',
    type: 'system_label',
    name: 'trash',
    unreadCount: 0,
    totalCount: 0
  }
  // {
  //   id: '5e892628d4bc60b4514d5d36',
  //   type: 'custom_label',
  //   name: 'Work',
  //   unreadCount: 1,
  //   totalCount: 1
  // },
  // {
  //   id: '5e8926820cf9ec6c834114ec',
  //   type: 'custom_label',
  //   name: 'Business',
  //   unreadCount: 0,
  //   totalCount: 2
  // }
];

export default labels;
