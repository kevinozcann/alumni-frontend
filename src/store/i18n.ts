import { persistReducer } from 'redux-persist';
import storage from 'redux-persist/lib/storage';
import { put, takeLatest } from 'redux-saga/effects';
import objectPath from 'object-path';
import { createSelector } from 'reselect';
import axios from 'axios';
import { GLOBAL_LOCALES_URL } from './ApiUrls';
import { ILanguage, TLang } from '../utils/shared-types';
import { IAction } from './store';

interface ILanguagesState {
  lang: TLang;
  languages: ILanguage[];
  error: string;
}
interface IParentLanguagesState {
  i18n: ILanguagesState;
}
type TActionAllState = ILanguagesState & {
  phase: string;
};

export const actionTypes = {
  I18N_SET_LANGUAGE: 'i18n/SET_LANGUAGE',
  I18N_PULL_LANGUAGES: 'i18n/PULL_LANGUAGES',
  I18N_SAVE_LANGUAGES: 'i18n/SAVE_LANGUAGES'
};

export const defaultLanguages: ILanguage[] = [
  {
    id: 1,
    title: 'Arabic',
    localTitle: 'العربية',
    shortCode: 'ar',
    longCode: 'ar-AE',
    icon: 'https://schst.in/arabic',
    svgIcon: 'saudi-arabia.svg'
  },
  {
    id: 2,
    title: 'English',
    localTitle: 'English',
    shortCode: 'en',
    longCode: 'en-US',
    icon: 'https://schst.in/english',
    svgIcon: 'united-states.svg'
  },
  {
    id: 3,
    title: 'German',
    localTitle: 'Deutsch',
    shortCode: 'de',
    longCode: 'de-DE',
    icon: 'https://schst.in/german',
    svgIcon: 'germany.svg'
  },
  {
    id: 4,
    title: 'Spanish',
    localTitle: 'español',
    shortCode: 'es',
    longCode: 'es-ES',
    icon: 'https://schst.in/spanish',
    svgIcon: 'spain.svg'
  },
  {
    id: 5,
    title: 'Turkish',
    localTitle: 'Türkçe',
    shortCode: 'tr',
    longCode: 'tr-TR',
    icon: 'https://schst.in/turkish',
    svgIcon: 'turkey.svg'
  }
];

const initialState: ILanguagesState = {
  lang: null,
  languages: defaultLanguages,
  error: null
};

export const i18nLangSelector = createSelector(
  (state: IParentLanguagesState) => objectPath.get(state, ['i18n', 'lang']),
  (lang: TLang) => lang
);

export const i18nLanguagesSelector = createSelector(
  (state: IParentLanguagesState) => objectPath.get(state, ['i18n', 'languages']),
  (i18nLanguages: ILanguage[]) => i18nLanguages
);

export const reducer = persistReducer(
  { storage, key: 'i18n', whitelist: ['i18n', 'lang', 'languages'] },
  (state: ILanguagesState = initialState, action: IAction<TActionAllState>): ILanguagesState => {
    switch (action.type) {
      case actionTypes.I18N_SET_LANGUAGE: {
        const { lang } = action.payload;
        return { ...state, lang };
      }
      case actionTypes.I18N_PULL_LANGUAGES: {
        return { ...state, error: null };
      }
      case actionTypes.I18N_SAVE_LANGUAGES: {
        const { languages } = action.payload;
        return { ...state, languages, error: null };
      }
      default:
        return state;
    }
  }
);

export const i18nActions = {
  setLanguage: (lang: TLang) => ({ type: actionTypes.I18N_SET_LANGUAGE, payload: { lang } }),
  pullLanguages: () => ({ type: actionTypes.I18N_PULL_LANGUAGES })
};

export function* saga() {
  yield takeLatest(actionTypes.I18N_PULL_LANGUAGES, function* pullLanguagesSaga() {
    // yield put(i18nActions.setLoginPhase('credentials-validating'));

    const response = yield axios.get(`${GLOBAL_LOCALES_URL}.json`);

    if (response.status !== 200) {
      // yield put(authActions.setLoginPhase('login-error', response.error));
      return;
    }

    const { data: languages } = response;

    yield put({
      type: actionTypes.I18N_SAVE_LANGUAGES,
      payload: { languages }
    });
  });
}
