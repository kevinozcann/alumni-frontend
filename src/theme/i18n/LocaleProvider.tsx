import React from 'react';
import { IntlProvider } from 'react-intl';
// import '@formatjs/intl-relativetimeformat/polyfill';
// import '@formatjs/intl-relativetimeformat/dist/locale-data/en';
// import '@formatjs/intl-relativetimeformat/dist/locale-data/tr';

import { TLang } from 'utils/shared-types';
import { ISchool } from 'pages/organization/organization-types';

import enMessages from './messages/en.json';

// const formats = {
//   number: {
//     TRY: {
//       style: 'currency',
//       currency: 'TRY'
//     },
//     USD: {
//       style: 'currency',
//       currency: 'USD'
//     }
//   }
// };

export type IntlMessageID = keyof typeof enMessages;

type TLocaleProviderProps = {
  lang: TLang;
  activeSchool: ISchool;
  children: React.ReactNode;
};

export const LocaleProvider = (props: TLocaleProviderProps) => {
  const { lang, activeSchool, children } = props;
  let messages = enMessages;

  React.useEffect(() => {
    if (lang) {
      import(`./messages/${lang}.json`)
        .then((module) => (messages = Object.assign(messages, module.default)))
        .then(() => {
          // Country overrides
          if (activeSchool?.config?.countryId?.toLowerCase() === 'tr') {
            import(`./overrides/country/tr/${lang}.json`)
              .then((module) => (messages = Object.assign(messages, module.default)))
              .catch(() => null);
          }

          // Institution Type overrides
          if (activeSchool?.config?.institutionType?.toLowerCase() === 'academy') {
            import(`./overrides/type/academy/${lang}.json`)
              .then((module) => (messages = Object.assign(messages, module.default)))
              .catch(() => null);
          }
        })
        .catch(() => null);
    }
  }, [activeSchool, lang]);

  return (
    // <IntlProvider locale={lang} defaultLocale='en' messages={messages} formats={formats}>
    <IntlProvider locale={lang} defaultLocale='en' messages={messages}>
      {children}
    </IntlProvider>
  );
};
