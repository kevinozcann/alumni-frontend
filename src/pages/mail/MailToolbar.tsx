import React from 'react';
import { useNavigate } from 'react-router-dom';
import { Box, IconButton, Tooltip } from '@mui/material';
import ArrowBackIcon from '@mui/icons-material/ArrowBack';

type MailToolbarProps = {
  section: string;
  label: string;
};

const MailToolbar = ({ section, label }: MailToolbarProps) => {
  const navigate = useNavigate();

  const handleBack = (): void => {
    if (label) {
      return navigate(`/mail/label/${label}`);
    }

    if (section) {
      return navigate(`/mail/${section}`);
    }

    return navigate('/mail/inbox');
  };

  return (
    <Box
      sx={{
        alignItems: 'center',
        backgroundColor: 'background.paper',
        display: 'flex',
        flexShrink: 0,
        height: 68,
        p: 2
      }}
    >
      <Tooltip title='Back'>
        <IconButton onClick={handleBack} size='large'>
          <ArrowBackIcon />
        </IconButton>
      </Tooltip>
      <Box sx={{ flexGrow: 1 }} />
      {/* <Hidden mdDown>
        <Box sx={{ width: 200 }}>
          <TextField
            fullWidth
            InputProps={{
              startAdornment: (
                <InputAdornment position='start'>
                  <SearchIcon fontSize='small' />
                </InputAdornment>
              )
            }}
            placeholder='Search message'
            size='small'
            variant='outlined'
          />
        </Box>
      </Hidden>
      <Tooltip title='Previous email'>
        <IconButton>
          <ChevronLeftIcon fontSize='small' />
        </IconButton>
      </Tooltip>
      <Tooltip title='Next email'>
        <IconButton>
          <ChevronRightIcon fontSize='small' />
        </IconButton>
      </Tooltip> */}
    </Box>
  );
};

export default MailToolbar;
