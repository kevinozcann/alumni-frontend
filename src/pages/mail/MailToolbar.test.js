import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

import { LocaleProvider } from 'theme/i18n/LocaleProvider';

import labels from './_store/labels';
import MailToolbar from './MailToolbar';

describe('MailToolbar component snapshot', () => {
  beforeEach(() => {
    jest.mock('react-router-dom', () => ({
      ...jest.requireActual('react-router-dom'),
      useNavigate: jest.fn()
    }));
  });

  it('should render as expected', () => {
    const tree = renderer
      .create(
        <LocaleProvider lang='en' activeSchool='1'>
          <MemoryRouter>
            <MailToolbar section='inbox' label={labels[0]} />
          </MemoryRouter>
        </LocaleProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
