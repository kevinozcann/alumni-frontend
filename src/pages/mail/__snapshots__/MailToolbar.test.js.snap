// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailToolbar component snapshot should render as expected 1`] = `
<div
  className="MuiBox-root css-1t8gvs6"
>
  <button
    aria-label="Back"
    aria-labelledby={null}
    className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeLarge css-mf1cb5-MuiButtonBase-root-MuiIconButton-root"
    data-mui-internal-clone-element={true}
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="ArrowBackIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
      />
    </svg>
    <span
      className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <div
    className="MuiBox-root css-i9gxme"
  />
</div>
`;
